app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js

```diff
@@ -95,341 +49,197 @@ class TableVisualization extends Visualization {
-        }
+        if (!this.tabElem) {
+            while (this.dom.firstChild) {
+                this.dom.removeChild(this.dom.lastChild)
+            }
 
-        .hiddenrows {
-            margin-left: 5px;
-            margin-top: 5px;
+            const style =
+                '.ag-theme-alpine { --ag-grid-size: 3px; --ag-list-item-height: 20px; display: inline; }'
+            const styleElem = document.createElement('style')
+            styleElem.innerHTML = style
+            this.dom.appendChild(styleElem)
+
+            const tabElem = document.createElement('div')
+            tabElem.setAttributeNS(null, 'id', 'vis-tbl-view')
+            tabElem.setAttributeNS(null, 'class', 'scrollable ag-theme-alpine')
+            this.dom.appendChild(tabElem)
+            this.tabElem = tabElem
+            this.updateTableSize()
+
+            this.agGridOptions = {
+                rowData: [],
+                columnDefs: [],
+                defaultColDef: {
+                    editable: false,
+                    sortable: true,
+                    filter: true,
+                    resizable: true,
+                    minWidth: 50,
+                },
+                enableRangeSelection: true,
+            }
+            this.agGrid = new agGrid.Grid(tabElem, this.agGridOptions)
         }
-        </style>`
 
-        const tabElem = document.createElement('div')
-        tabElem.setAttributeNS(null, 'id', 'vis-tbl-view')
-        tabElem.setAttributeNS(null, 'class', 'scrollable')
-        tabElem.setAttributeNS(null, 'width', '100%')
-        tabElem.setAttributeNS(null, 'height', '100%')
-        this.tabElem = tabElem
-        this.dom.appendChild(tabElem)
-        this.updateTableSize()
+        let parsedData = typeof data === 'string' ? JSON.parse(data) : data
 
-        let parsedData = data
-        if (typeof data === 'string') {
-            parsedData = JSON.parse(data)
-        }
-
-        let style = style_light
-        if (document.getElementById('root').classList.contains('dark-theme')) {
-            style = style_dark
-        }
+        let columnDefs = []
+        let rowData = []
+        let dataTruncated = false
 
         if (parsedData.error !== undefined) {
-            tabElem.innerHTML = 'Error: ' + parsedData.error
+            this.agGridOptions.api.setColumnDefs([
+                {
+                    field: 'error',
+                    width: 1000,
+                    cellStyle: { 'white-space': 'normal' },
+                },
+            ])
+            this.agGridOptions.api.setRowData([{ error: parsedData.error }])
+        } else if (parsedData.json !== undefined && isMatrix(parsedData.json)) {
+            columnDefs = parsedData.json[0].map((_, i) => ({ field: i.toString() }))
+            rowData = parsedData.json
+            dataTruncated = parsedData.all_rows_count !== parsedData.json.length
+        } else if (parsedData.json !== undefined && isObjectMatrix(parsedData.json)) {
```

Comment:
use `!= null` instead of `!== undefined`. The former covers the latter as well in JS and is just more generic / safe.